### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by NasyTheSlayer**
[DEMO LINK](https://NasyTheSlayer.github.io/react_todo-app-loading-todos/)

### CODE REVIEW

**Review by Esceype — CHANGES_REQUESTED** [review]
GJ!
Need to make some fixes in edit mode:
- after edit with spaces it shows okay, but if open the same todo after editing it saves the spaces in end on in start, you need to trim it before save
-  if I save empty todo I see the placeholder, but it doesn`t work, if I try to save empty todo it save previous value and not delete it.

File: src/components/Eror/Error.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import React, { useContext } from 'react';
+import cn from 'classnames';
+import { TodosContext } from '../TodosContext/TodosContext';
+
+export const Error: React.FC = () => {
+  const { errorMessage, setErrorMessage } = useContext(TodosContext);
+  let isErrorHidden = true;
+
+  if (errorMessage !== '') {
+    isErrorHidden = false;
```
Comment (Andriy-Shymkiv) [inline src/components/Eror/Error.tsx:11]: create state for this

File: src/components/Footer/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+export const Footer = () => {
+  const { todos, setTodos } = useContext(TodosContext);
+
+  const handleDeleteCompleted = () => {
+    const newTodos = todos.filter(todo => !todo.completed);
+
+    setTodos(newTodos);
+  };
+
+  const isVisibleCrealButton = todos.some(todo => todo.completed);
```
Comment (Andriy-Shymkiv) [inline src/components/Footer/Footer.tsx:15]: what is `Creal`?

File: src/utils/getNumberActiveTodo.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  let number = 0;
+
+  todos.forEach(todo => {
+    if (!todo.completed) {
+      number += 1;
+    }
+  });
+
+  return number;
+};
```
Comment (Andriy-Shymkiv) [inline src/utils/getNumberActiveTodo.tsx:13]:
```suggestion
export const getActiveTodosLength = (todos: Todo[]) => {
  return todos.filter(todo => !todo.completed).length
};
```

**Review by andriy-shymkiv — APPROVED** [review]
good job. left some comments. approved in advance

**Pull request closed without merging**